components/layout_2020/replaced.rs

```diff
@@ -82,25 +118,83 @@ impl ReplacedContent {
             .max_box_size()
             .percentages_relative_to(containing_block);
 
-        let clamp = |inline_size: Length, block_size: Length| {
-            (
-                inline_size.clamp_between_extremums(min_box_size.inline, max_box_size.inline),
-                block_size.clamp_between_extremums(min_box_size.block, max_box_size.block),
-            )
+        let default_object_size = || {
+            // FIXME:
+            // “If 300px is too wide to fit the device, UAs should use the width of
+            //  the largest rectangle that has a 2:1 ratio and fits the device instead.”
+            // “height of the largest rectangle that has a 2:1 ratio, has a height not greater
+            //  than 150px, and has a width not greater than the device width.”
+            physical::Vec2 {
+                x: Length::new(300.),
+                y: Length::new(150.),
+            }
+            .size_to_flow_relative(mode)
+        };
+        let clamp = |inline_size: Length, block_size: Length| Vec2 {
+            inline: inline_size.clamp_between_extremums(min_box_size.inline, max_box_size.inline),
+            block: block_size.clamp_between_extremums(min_box_size.block, max_box_size.block),
         };
         // https://drafts.csswg.org/css2/visudet.html#min-max-widths
         // https://drafts.csswg.org/css2/visudet.html#min-max-heights
-        let (inline_size, block_size) = match (box_size.inline, box_size.block) {
+        match (box_size.inline, box_size.block) {
             (LengthOrAuto::LengthPercentage(inline), LengthOrAuto::LengthPercentage(block)) => {
                 clamp(inline, block)
             },
             (LengthOrAuto::LengthPercentage(inline), LengthOrAuto::Auto) => {
-                clamp(inline, inline / intrinsic_ratio)
+                let block = if let Some(i_over_b) = intrinsic_ratio {
```

Comment:
This name confuses me, please just name it `ratio`.

Reply:
The problem these new names try to address is that when A divided by B is called a ratio, B divided by A is also called a ratio although they’re not interchangeable. Code that uses that ratio sometimes to multiple and sometimes to divide a length, having a variable name that indicates the “direction” of the ratio helps check if those are correct by doing pseudo-dimensional analysis.

We could declare that an aspect ratio is by convention always a width divided by a height. But this code work in flow-relative dimensions and accesses a ratio field in physical dimensions. I’ve added `inline_size_over_block_size_intrinsic_ratio` to make that conversion.

I’m open to other names, but I’d prefer if they indicate the “direction” of the ratio somehow. Or maybe we could provide that indication in doc-comments and code comments?

Reply:
Fair enough, I don't mind the name.